Assignment1-JavaScript/Queue_via_Stacks.js

```diff
@@ -0,0 +1,36 @@
+class Queue {
+    constructor (){
+        this.stack1 = []; //push
+        this.stack2 = []; //pops
+    }
+    //enqueue 
+    enqueue(element){
+        this.stack1.push(element);
+    }
+    //dequeue
+    dequeue(){
+        if (this.stack2.length == 0){
+            while (this.stack1.length > 0){
+                this.stack2.push(this.stack1.pop());
+            }
+        }
+        if (this.stack2.length === 0){
+            throw new Error("Empty Queue");
+        }
+        return this.stack2.pop();
+    }
+    isEmpty() {
+        return this.stack1.length === 0 &&this.stack2.length === 0;
+    }
+
+}
+// Test
```

Comment:
Nice validation! Checkout my JS recitation vids to use mocha for testing rather than just console.log()'s!